src/engine/sparqlExpressions/LangExpression.h

```diff
@@ -55,6 +58,8 @@ inline auto getLanguageTag = [](OptValue optLangTag) -> IdOrLiteralOrIri {
 
 }  //  namespace detail::langImpl
 
+// TODO: Directly implement this in a .cpp file and make it
+// available as makeLangExpression.
```

Comment:
I figured out what you have to do:
You completely move it to a `.cpp` file,
And the only function that is needed, which is not part of the virtual interface is the `Variable` function.
But you can expose this via a (standalone)

`std::optional<Varaiable> getVariableFromLangExpression()` function, which is defined in the same `.cpp` file, and thus has access to the implementation.
I think this is doable for you, please contact me, if you have questions.

Reply:
Yes, i think i got it. The function `getVariableFromLanfExpression()` is visible over `NaryExpression.h` + `LangExpression` is now used over `makeLangExpression(SparqlExpression....)`.